hs_composite_resource/models.py

```diff
@@ -138,35 +362,73 @@ def supports_logical_file(self):
         return True
 
     def get_metadata_xml(self, pretty_print=True, include_format_elements=True):
-        from lxml import etree
-
         # get resource level core metadata as xml string
         # for composite resource we don't want the format elements at the resource level
         # as they are included at the aggregation map xml document
-        xml_string = super(CompositeResource, self).get_metadata_xml(pretty_print=False,
+        xml_string = super(CompositeResource, self).get_metadata_xml(pretty_print=True,
                                                                      include_format_elements=False)
+        return xml_string
 
-        # create an etree xml object
-        RDF_ROOT = etree.fromstring(xml_string)
-
-        return etree.tostring(RDF_ROOT, encoding='UTF-8', pretty_print=pretty_print).decode()
-
-    def _recreate_fileset_xml_docs(self, folder):
-        """Recreates xml files for all fileset aggregations that exist under the path 'folder'
-        as well as for any parent fileset that may exist relative to path 'folder'
+    def _recreate_nested_aggr_xml_docs(self, folder, nested_aggr):
+        """Recreates xml files for all fileset or model instance aggregations that exist under the path 'folder'
+        as well as for any parent fileset/model instance that may exist relative to path 'folder'
         """
+        if nested_aggr == 'fileset':
+            nested_aggr_set = self.filesetlogicalfile_set
+            aggr_in_path_func = self.get_fileset_aggregation_in_path
+        else:
+            # create xml files for all model instance aggregation that may exist under *folder*
+            nested_aggr_set = self.modelinstancelogicalfile_set
+            aggr_in_path_func = self.get_model_aggregation_in_path
 
-        filesets = self.filesetlogicalfile_set.filter(folder__startswith=folder)
-        for fs in filesets:
-            fs.create_aggregation_xml_documents()
+        nested_aggrs = nested_aggr_set.filter(folder__startswith=folder)
+        for ns_aggr in nested_aggrs:
+            ns_aggr.create_aggregation_xml_documents()
 
-        # Also need to recreate xml doc for any parent fileset that may exist relative to path
+        # Also need to recreate xml doc for any parent fileset/model instance that may exist relative to path
         # *folder*
         if '/' in folder:
             path = os.path.dirname(folder)
-            parent_fs = self.get_fileset_aggregation_in_path(path)
-            if parent_fs is not None:
-                parent_fs.create_aggregation_xml_documents()
+            parent_aggr = aggr_in_path_func(path)
+            if parent_aggr is not None:
+                parent_aggr.create_aggregation_xml_documents()
+
+    def create_model_program_meta_json_schema_files(self, path=''):
```

Comment:
Just reading this method name suggest this logic belongs in the logical files, not composite resource.  Is there any reason we the model program can't create these json schema files everytime the create_aggregation_xml_files is called?

Reply:
Creation of schema file is now part of the xml file creation function